src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.tsx

```diff
@@ -14,30 +14,36 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from 'src/store';
 
 import {
   addSelectedSpecies,
   removeSelectedSpecies
 } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
 import { getSelectedSpeciesList } from 'src/content/app/tools/blast/state/blast-form/blastFormSelectors';
+import { getPopularSpecies } from 'src/content/app/species-selector/state/speciesSelectorSelectors';
+import { fetchPopularSpecies } from 'src/content/app/species-selector/state/speciesSelectorSlice';
 
 import Checkbox from 'src/shared/components/checkbox/Checkbox';
 
-import speciesList from './speciesList';
-
 import type { Species } from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
 
 import styles from './BlastSpeciesSelector.scss';
 
 const BlastSpeciesSelector = () => {
-  const dispatch = useDispatch();
+  const dispatch = useAppDispatch();
+  const popularSpeciesList = useSelector(getPopularSpecies);
   const selectedSpeciesList = useSelector(getSelectedSpeciesList);
   const selectedGenomeIds = selectedSpeciesList.map(
     ({ genome_id }) => genome_id
   );
 
+  useEffect(() => {
```

Comment:
why do we need to dispatch this?

Reply:
If a user come to blast view first, we need to fetch the popular species

Reply:
Then should the useEffect not have `popularSpeciesList` as a check. My understanding of useEffect and the way its written here is everytime the component is loaded it will fetch the species. And we only want to fetch it when there is no species.